third_party/proton/proton/viewer.py

```diff
@@ -92,18 +92,20 @@ def get_min_time_bytes(df, device_info):
 def derive_metrics(gf, metrics, raw_metrics, device_info):
     derived_metrics = []
     original_metrics = []
-    time_metric_name = match_available_metrics([time_factor_dict.name], raw_metrics)[0]
-    time_unit = (time_factor_dict.name + "/" + time_metric_name.split("(")[1].split(")")[0])
+    internal_frame_indices = gf.dataframe["DeviceId"].isna()
 
     def get_time_seconds(df):
+        time_metric_name = match_available_metrics([time_factor_dict.name], raw_metrics)[0]
+        time_unit = (time_factor_dict.name + "/" + time_metric_name.split("(")[1].split(")")[0])
         return df[time_metric_name] * time_factor_dict.factor[time_unit]
 
     for metric in metrics:
         if metric == "util":  # Tensor core only
             min_time_bytes = get_min_time_bytes(gf.dataframe, device_info)
             min_time_flops = get_min_time_flops(gf.dataframe, device_info)
-            time_sec = get_time_seconds(gf.dataframe, time_metric_name, time_unit)
+            time_sec = get_time_seconds(gf.dataframe)
```

Comment:
Good catch, i missed that. I guess I should write unit tests for the other metrics to avoid this